### PR TITLE
List Exports: angular => knockout

### PR DESCRIPTION
**Pull request by orangejenny**
Note that this is a PR into https://github.com/dimagi/commcare-hq/pull/22128

@esoergel @nickpell this is lower priority for review than other currently-open PRs, since it's going to wait on QA.

[QA ticket](https://manage.dimagi.com/default.asp?284978)

### CODE REVIEW

**Comment by orangejenny** [comment]
new code buddy @jmtroth0 could you review this one?

**Review by jmtroth0 — APPROVED** [review]
looks good, although i mostly just skimmed it. the comments are random things i saw along the way and definitely aren't blocking

File: corehq/apps/export/static/export/js/export_list.js [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+        self.endDate = ko.observable().extend({
+            pattern: {
+                message: gettext('Invalid date format'),
+                params: '^[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]$',
+            },
+        });
+
+        // Editing filters for a saved export
+        self.selectedExportModelType = ko.observable();
+        self.filterModalExportId = ko.observable();
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/export_list.js]: yes - they're all updated each time a user clicks an "Edit Filters" button, to match the values for whichever export is being edited

File: corehq/apps/export/static/export/js/export_list_main.js [hunk 2]
```diff
@@ -28,22 +28,48 @@ hqDefine("export/js/export_list_main", function () {
...
+                // Determine whether or not to show bulk export download button & message
+                if (newValue !== self.showBulkExportDownload()) {
+                    self.showBulkExportDownload(!!_.find(self.exports, function (e) {
+                        return e.addedToBulk();
+                    }));
+                }
+
+                // Update hidden value of exports to download
+                if (self.showBulkExportDownload()) {
+                    self.bulkExportList(JSON.stringify(_.map(_.filter(self.exports, function (e) {
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/export_list_main.js]: Moved this logic into a function that's only called when the bulk export form is submitted. And updated names (tangentially related, I really wish `export` weren't a reserved word).

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/export_list_main.js]: (9f8c67513aa)

File: corehq/apps/export/static/export/js/export_list.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        self.endDateHasError = ko.computed(function () {
+            return !self.endDate.isValid();
+        });
+        self.disableSubmit = ko.computed(function () {
+            return self.showStartDate() && self.startDateHasError()
+                || self.showEndDate() && self.endDateHasError();
+        });
+        self.commitFilters = function () {
+            var export_ = _.find(self.exports, function (e) { return e.id() === self.filterModalExportId(); });
+            self.isSubmittingForm(true);
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/export_list.js]: Verified that there's a [server-side check](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/export/views/list.py#L519-L522) so even if there are any js issues, the form submission will fail and the user will be notified.

**Pull request merged**